libs/SalesforceSDK/src/com/salesforce/androidsdk/config/BootConfig.java

```diff
@@ -98,6 +98,49 @@ public static BootConfig getBootConfig(Context ctx) {
 		}
 		return INSTANCE;
 	}
+
+	/**
+	 * Validates a boot config's inputs against basic sanity tests.
+	 * @param config The BootConfig instance to validate.
+	 * @return A BootConfigValidationResult describing the success or failure of validation, with
+	 * an the error message associated with a failed validation.
+	 */
+	public static BootConfigValidationResult validateBootConfig(BootConfig config) {
```

Comment:
The logic was getting a little crazy, trying to juxtapose valid URL values against local/remote configs and authentication, in tandem.  This sanity checks all of the conditions that aren't subject to workflow.